server/proxy/proxy.go

```diff
@@ -178,3 +187,67 @@ func (p *Proxy) upstreamForApp(app *apps.App) (upstream.Upstream, error) {
 		return nil, utils.NewInvalidError("not a valid app type: %s", app.AppType)
 	}
 }
+
+func (p *Proxy) CleanUserCallContext(userID string, cc *apps.Context) (*apps.Context, error) {
+	cc = &apps.Context{
+		ContextFromUserAgent: cc.ContextFromUserAgent,
+	}
```

Comment:
Unwanted fields are discarded here by redefining the context as just the user agent fields